ThermodynamicsSA.pyx

```diff
@@ -267,27 +323,41 @@ cdef class ThermodynamicsSA:
             Py_ssize_t jmax = Gr.dims.nlg[1]
             Py_ssize_t kmax = Gr.dims.nlg[2]
             Py_ssize_t count
-            Py_ssize_t s_shift = PV.get_varshift(Gr, 's')
+            Py_ssize_t s_shift
             Py_ssize_t qt_shift = PV.get_varshift(Gr, 'qt')
             double[:] data = np.empty((Gr.dims.npg,), dtype=np.double, order='c')
             double[:] tmp
 
 
-
-        # Ouput profiles of thetas
-        with nogil:
-            count = 0
-            for i in range(imin, imax):
-                ishift = i * istride
-                for j in range(jmin, jmax):
-                    jshift = j * jstride
-                    for k in range(kmin, kmax):
-                        ijk = ishift + jshift + k
-                        data[count] = thetas_c(PV.values[s_shift + ijk], PV.values[qt_shift + ijk])
-
-                        count += 1
-
-
+        #If entropy is not a prognostic variable get it from the diagnostic variable class
+        if 's' in PV.name_index:
+            s_shift = PV.get_varshift(Gr, 's')
+            # Ouput profiles of thetas
+            with nogil:
+                count = 0
+                for i in range(imin, imax):
+                    ishift = i * istride
+                    for j in range(jmin, jmax):
+                        jshift = j * jstride
+                        for k in range(kmin, kmax):
+                            ijk = ishift + jshift + k
+                            data[count] = thetas_c(PV.values[s_shift + ijk], PV.values[qt_shift + ijk])
+
+                            count += 1
+        else:
+            s_shift = DV.get_varshift(Gr, 's')
+            # Ouput profiles of thetas
+            with nogil:
+                count = 0
+                for i in range(imin, imax):
+                    ishift = i * istride
+                    for j in range(jmin, jmax):
+                        jshift = j * jstride
+                        for k in range(kmin, kmax):
+                            ijk = ishift + jshift + k
+                            data[count] = thetas_c(DV.values[s_shift + ijk], PV.values[qt_shift + ijk])
+
+                            count += 1
```

Comment:
I guess this is supposed to be related to thetali, but it seems exactly the same as the code below if s in PV.name_index.

Reply:
are you suggesting to remove this ?

Reply:
Yes.